site/app/controllers/forum/ForumController.php

```diff
@@ -369,6 +370,60 @@ public function publishThread() {
         return $this->core->getOutput()->renderJsonSuccess($result);
     }
 
+    /**
+     * @Route("/courses/{_semester}/{_course}/forum/make_announcement", methods={"POST"})
+     * @AccessControl(permission="forum.modify_announcement")
+     */
+    public function makeAnnouncement(): JsonResponse {
+        if (!isset($_POST['id'])) {
+            $this->core->addErrorMessage("thread_id not provided");
+            return JsonResponse::getFailResponse("thread_id not provided");
+        }
+        // Check that the post is the first post of the thread
+        $thread_info = $this->core->getQueries()->findPost($_POST['id']);
+        if ($thread_info["parent_id"] != -1) {
```

Comment:
I think a PHP warning will be raised here if `$_POST['id']` does not point at an actual thread.

Reply:
Though this query / check should be rewritten so that it also checks within the query that `parent_id = -1` as currently the query within `findPost` will return all posts within a thread (which could be a large amount, and waste memory to fetch) and that there's no guarantee that the first result in the result set will be the top post post in a given thread.